### ISSUE TITLE
codefix: Remove declaration for local in destructuring assignment breaks code.

### ISSUE THREAD

**Issue by NaridaL** [Microsoft/TypeScript#22330]
**TypeScript Version:**  2.8.0-dev.20180302

**Search Terms:** `"remove declaration for"` `remove declaration destructuring`

**Code**

--noUnusedLocals needs to be on.
The "remove declaration for" quick fix for unused locals is broken:

```ts
function f() {
	const {abs, floor, ceil} = Math
	const {sin, cos, tan} = Math
}
```

**Expected behavior:**
After code fix on `abs`:
```ts
function f() {
	const {floor, ceil} = Math
	const {sin, cos, tan} = Math
}
```
After code fix "fix all":
```ts
function f() {
}
```
Possible side effects of the RHS are ignored by the `const abs = Math.abs` quick fix, so this should be mirrored here.
**Actual behavior:**
```ts
function f() {
	const {, floor, ceil} = Math
	const {sin, cos, tan} = Math
}
```
After code fix "fix all":
```ts
function f() {
	const {,,} = Math
	const {,,} = Math
}
```

screencast: https://gfycat.com/SleepyEnchantingArcticfox

**Related Issues:**

**Comment by mhegazy** [comment]
PRs welcomed!

**Comment by Kingwl** [comment]
IMO `noUnusedLocals ` need some change: 
error on (parameter, variable declaration)list node if all item is unused

**Comment by andy-ms** [comment]
Fixed by #23805 and #23819
